crates/typst-layout/src/flow/distribute.rs

```diff
@@ -324,10 +325,28 @@ impl<'a, 'b> Distributor<'a, 'b, '_, '_, '_> {
         &mut self,
         frame: Frame,
         align: Axes<FixedAlignment>,
-        sticky: bool,
+        sticky: Option<Sticky>,
         breakable: bool,
     ) -> FlowResult<()> {
-        if sticky {
+        // Check if the successor frame has sticky: above
+        let has_sticky_successor = self
+            .composer
+            .work
+            .children
+            .iter()
+            .skip(1)
+            .find_map(|child| match child {
+                Child::Single(single) => Some(
+                    single.sticky.as_ref().map(|s| s.is_sticky_above()).unwrap_or(false),
+                ),
+                Child::Multi(multi) => Some(
+                    multi.sticky.as_ref().map(|s| s.is_sticky_above()).unwrap_or(false),
+                ),
+                _ => None,
+            })
+            .unwrap_or(false);
+
+        let mut stick_to_successor = || {
```

Comment:
I'd love some tests showing how a sequence of sticky: "above" which don't fit in a single page behave. There is one already for sticky: "below", not too sure if it was `block-sticky-many`, but might have been